### PR TITLE
Sublime syntax highlighter

### PR DESCRIPTION
**Pull request by kvark**
It's really crappy and needs more love, but sufficient for my basic needs :) Any help improving it would be great! Example:
ron-highlight

### CODE REVIEW

File: highlight/RON.sublime-syntax [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+      push:
+        - meta_scope: comment.block.ron
+        - match: \*/
+          pop: true
+    - match: (//).*$\n?
+      scope: comment.line.double-slash.js
+      captures:
+        1: punctuation.definition.comment.ron
+  constant:
+    - match: \b(?:true|false)\b
```
Comment (torkleyy) [inline highlight/RON.sublime-syntax]: What does `?:` do? Wouldn't it work without?

File: highlight/RON.sublime-syntax [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+            (?:
+              [eE]   # an e character
+              [+-]?  # followed by an option +/-
+              \d+    # followed by one or more digits
+            )?       # make exponent optional
+          )?         # make decimal portion optional
+        )
+      scope: constant.numeric.ron
+  object:
+    - match: '[A-Z][a-zA-Z]*'
```
Comment (torkleyy) [inline highlight/RON.sublime-syntax]: I'm not sure if we want to enforce capital letters by syntax rules. Also, this doesn't allow `Ipv4Addr`, does it?

File: highlight/RON.sublime-syntax [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+  dictionary:
+    - match: '\{'
+      scope: punctuation.section.dictionary.begin.ron
+      push:
+        - meta_scope: meta.structure.dictionary.ron
+        - match: '\}'
+          scope: punctuation.section.dictionary.end.ron
+          pop: true
+        # only string keys are supported by this highlighter atm
+        - match: '"'
```
Comment (torkleyy) [inline highlight/RON.sublime-syntax]: Couldn't this just include `value`?

File: highlight/RON.sublime-syntax [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+          scope: punctuation.definition.string.begin.ron
+          push:
+            - meta_scope: meta.structure.dictionary.key.ron string.quoted.double.ron
+            - include: inside-string
+        - include: comments
+        - match: ':'
```
Comment (torkleyy) [inline highlight/RON.sublime-syntax]: Later this this context should be pushed as soon as a value (the key) has been parsed, right?

**Comment by kvark** [comment]
Thanks for a quick look! I addressed the concerns now, hopefully ;)

File: highlight/RON.sublime-syntax [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+            (?:
+              [eE]   # an e character
+              [+-]?  # followed by an option +/-
+              \d+    # followed by one or more digits
+            )?       # make exponent optional
+          )?         # make decimal portion optional
+        )
+      scope: constant.numeric.ron
+  object:
+    - match: '[A-Z][0-9a-zA-Z]*'
```
Comment (torkleyy) [inline highlight/RON.sublime-syntax]: This shouldn't disallow a lowercase first character.

File: highlight/RON.sublime-syntax [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+      scope: entity.name.class.ron
+    - match: '\('
+      scope: punctuation.section.dictionary.begin.ron
+      push:
+        - meta_scope: meta.structure.entity.ron
+        - match: '\)'
+          scope: punctuation.section.dictionary.end.ron
+          pop: true
+        - match: '[a-z_][a-zA-Z0-9_]*'
```
Comment (torkleyy) [inline highlight/RON.sublime-syntax]: And tbis shouldn't disallow an uppercase first character. I'd suggest you add an `ident` and use it for both the struct name and the field names.

**Comment by kvark** [comment]
@torkleyy ok, third time is the charm ;)

**Pull request merged**